src/graph/query_graph.c

```diff
@@ -105,6 +93,16 @@ void QueryGraph_AddNode(QueryGraph *qg, QGNode *n) {
 	qg->nodes = array_append(qg->nodes, n);
 }
 
+void QueryGraph_SetNodeLabel(QGNode *n, const char *label) {
+	GraphContext *gc = QueryCtx_GetGraphCtx();
```

Comment:
Now that this function is exposed, I think we should `ASSERT(n->labelID == GRAPH_NO_LABEL && label != NULL)`
where `ASSERT` is the new debug only assertion see #1265